src/vse_sync_pp/analyzers/pmc.py

```diff
@@ -0,0 +1,134 @@
+### SPDX-License-Identifier: GPL-2.0-or-later
+
+"""Analyze PMC log messages"""
+
+from .analyzer import Analyzer
+import copy
+
+STATE_FREERUN = 248
+STATE_LOCKED = 6
+STATE_HOLDOVER_IN_SPEC = 7
+STATE_HOLDOVER_OUT_OF_SPEC1 = 140
+STATE_HOLDOVER_OUT_OF_SPEC2 = 150
+STATE_HOLDOVER_OUT_OF_SPEC3 = 160
+
+STATE_NAMES = {
+   STATE_FREERUN: "FREERUN",
+   STATE_LOCKED: "LOCKED",
+   STATE_HOLDOVER_IN_SPEC: "HOLDOVER_IN_SPEC",
+   STATE_HOLDOVER_OUT_OF_SPEC1: "HOLDOVER_OUT_SPEC1",
+   STATE_HOLDOVER_OUT_OF_SPEC2: "HOLDOVER_OUT_SPEC2",
+   STATE_HOLDOVER_OUT_OF_SPEC3: "HOLDOVER_OUT_SPEC3",
+}
+
+STATE_TRANSITION = {
+    STATE_FREERUN: [STATE_FREERUN, STATE_LOCKED],
+    STATE_LOCKED: [STATE_LOCKED, STATE_HOLDOVER_IN_SPEC],
+    STATE_HOLDOVER_IN_SPEC: [STATE_LOCKED, 
+                             STATE_HOLDOVER_IN_SPEC, 
+                             STATE_HOLDOVER_OUT_OF_SPEC1, 
+                             STATE_HOLDOVER_OUT_OF_SPEC2, 
+                             STATE_HOLDOVER_OUT_OF_SPEC3],
+    STATE_HOLDOVER_OUT_OF_SPEC1: [STATE_LOCKED, 
+                                  STATE_HOLDOVER_OUT_OF_SPEC1, 
+                                  STATE_HOLDOVER_OUT_OF_SPEC2, 
+                                  STATE_HOLDOVER_OUT_OF_SPEC3],
+    STATE_HOLDOVER_OUT_OF_SPEC2: [STATE_LOCKED, 
+                                  STATE_HOLDOVER_OUT_OF_SPEC1, 
+                                  STATE_HOLDOVER_OUT_OF_SPEC2, 
+                                  STATE_HOLDOVER_OUT_OF_SPEC3],
+    STATE_HOLDOVER_OUT_OF_SPEC3: [STATE_LOCKED, 
+                                  STATE_HOLDOVER_OUT_OF_SPEC1, 
+                                  STATE_HOLDOVER_OUT_OF_SPEC2, 
+                                  STATE_HOLDOVER_OUT_OF_SPEC3],
+}
+
+BASE_CLOCK_CLASS_COUNT = {
+    "count": 0,
+    "transitions": {
+        STATE_FREERUN: 0,
+        STATE_LOCKED: 0,
+        STATE_HOLDOVER_IN_SPEC: 0,
+        STATE_HOLDOVER_OUT_OF_SPEC1: 0,
+        STATE_HOLDOVER_OUT_OF_SPEC2: 0,
+        STATE_HOLDOVER_OUT_OF_SPEC3: 0,
+    },
+}
+
+def is_transition_allowed(current_state, new_state):
+   return new_state in STATE_TRANSITION[current_state]
+
+def get_named_clock_class_result(clock_class_count):
+    named_clock_class_count = {STATE_NAMES[k]:v for (k,v) in clock_class_count.items()}
+    for clock_class in clock_class_count.values():
+        clock_class["transitions"] = {STATE_NAMES[k]:v for (k,v) in clock_class["transitions"].items()}
+    return named_clock_class_count
+
+class ClockStateAnalyzer(Analyzer):
+    """Analyze clock state
+    """
+    id_ = 'phc/gm-settings'
+    parser = id_
+    def __init__(self, config):
+        super().__init__(config)
+        # minimum test duration for a valid test
+        self._duration = config.parameter('min-test-duration/s')
+        self.state = None
```

Comment:
Is this an instance variable? Only appears in method test?

Reply:
Good point. Originally I wanted to use it in other places as well.
Moved to the test method